proposal/calc-functions.md

```diff
@@ -0,0 +1,363 @@
+  * If `number`'s value is negative, return `-1`.
+  * Otherwise, return a unitless number with the same value as `number`.
+  
+    > In this case, `number` is either positive zero, negative zero, or NaN.
+  
+  > To match CSS's behavior, these computations _don't_ use fuzzy comparisons.
+
+* If `calc`'s name is `"atan2"`, `"pow"`, or `"log"`:
+
+  * If `calc`'s name isn't `"log"`, `arguments` has only one element, and it's
+    not an unquoted string or a `CalculationInterpolation`, throw an error.
+  
+  * Otherwise, if `arguments` contains exactly two numbers with [known units],
+    return the result of passing that number to the function in [`sass:math`]
+    whose name matches `calc`'s.
+
+* If `calc`'s name is `"mod"` or `"rem"`:
+
+  * If `arguments` has only one element and it's not an unquoted string or a
+    `CalculationInterpolation`, throw an error.
+
+  * Otherwise, if `arguments` contains exactly two numbers `dividend` and
+    `modulus`:
+
+    > These numbers need not have known units, since mod and rem are linear
+    > functions so they'll work equivalently for percentages as for the numbers
+    > they represent.
+
+    * If `dividend` and `modulus` are [definitely-incompatible], throw an error.
+
+      [definitely-incompatible]: ../spec/types/number.md#possibly-compatible-numbers
+
+    * If `dividend` and `modulus` are mutually [compatible], let `result` be the
+      result of `dividend % modulus`.
+
+      [compatible]: ../spec/types/number.md#compatible-units
+
+    * If `calc`'s name is `"rem"`, and if `dividend` is positive and `modulus`
+      is negative or vice versa, set `result` to `mod - dividend`.
+
+    * Return `result`.
+
+* If `calc`'s name is `"round"`:
+
+  * If `arguments` has exactly three elements, set `strategy`, `number`, and
+    `step` to those arguments respectively.
+
+  * Otherwise, if `arguments` has exactly two elements:
+
+    * If the first element is an unquoted string or interpolation with value
+      `"nearest"`, `"up"`, `"down"`, or `"to-zero"`, and the second argument
+      isn't an unquoted string or `CalculationInterpolation`, throw an error.
+
+      > Normally we allow unquoted strings anywhere in a calculation, but this
+      > helps catch the likely error of a user accidentally writing `round(up,
+      > 10px)` without realizing that it needs a third argument.
+  
+    * Otherwise, set `number` and `step` to the two arguments respectively and
+      `strategy` to an unquoted string with value `"nearest"`.
+
+  * Otherwise, if the single argument isn't an unquoted string or
+    `CalculationInterpolation`, throw an error.
+
+  * If `strategy`, `number`, and `step` are set:
+
+    * If `strategy` isn't a [special variable string], nor is it an unquoted
+      string or interpolation with value `"nearest"`, `"up"`, `"down"`, or
+      `"to-zero"`, throw an error.
+
+    * If `strategy` is an unquoted string or interpolation and both `number` and
```

Comment:
That's right. Everything just falls through to that unless explicitly specified otherwise, because there are a bunch of cases that can only be handled in the browser because they require rendering knowledge to resolve.